### PR TITLE
Adding CS0267 to OrderModifiers FixableDiagnosticIds

### LINKED ISSUE #30352: OrderModifiers should say it fixes CS0267 (The 'partial' modifier can only appear immediately before 'class', 'struct', 'interface', or 'void')
OrderModifiers can currently fix CS0267 (The 'partial' modifier can only appear immediately before 'class', 'struct', 'interface', or 'void'):
```c#
partial abstract class C { }
```
to
```c#
abstract partial class C { }
```
but it doesn't say so using `FixableDiagnosticIds`. This means there is no error-bulb and the fix isn't discoverable. Adding it might also have other benefits such as proper prioritization of this fix (but I'm not sure where this would have an effect).

### PR DESCRIPTION
**Pull request by Neme12**
Fixes #30352 

**Before:**
previous2

**After:**
new2

### CODE REVIEW

File: src/Features/VisualBasic/Portable/OrderModifiers/VisualBasicOrderModifiersCodeFixProvider.vb [hunk 1]
```diff
@@ -15,5 +16,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.OrderModifiers
                        VisualBasicCodeStyleOptions.PreferredModifierOrder,
                        VisualBasicOrderModifiersHelper.Instance)
         End Sub
+
+        Protected Overrides ReadOnly Property FixableCompilerErrorIds As ImmutableArray(Of String) =
+            ImmutableArray(Of String).Empty
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/OrderModifiers/VisualBasicOrderModifiersCodeFixProvider.vb]: might warrant an explanation.

File: src/EditorFeatures/CSharpTest/OrderModifiers/OrderModifiersTests.cs [hunk 2]
```diff
@@ -274,5 +274,45 @@ internal static class C
...
+        {
+            await TestInRegularAndScript1Async(
+@"[|partial|] static class C { }",
+@"static partial class C { }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsOrderModifiers)]
+        public async Task PartialAtTheEnd5()
+        {
+            await TestInRegularAndScript1Async(
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/OrderModifiers/OrderModifiersTests.cs]: how about adding just sanity simple test for struct and interface if that is not too much trouble?

also, does this also handle parital method case as well?

Comment (Neme12) [reply src/EditorFeatures/CSharpTest/OrderModifiers/OrderModifiersTests.cs]: I will add the tests.

File: src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersCodeFixProvider.cs [hunk 3]
```diff
@@ -16,5 +17,9 @@ public CSharpOrderModifiersCodeFixProvider()
             : base(CSharpSyntaxFactsService.Instance, CSharpCodeStyleOptions.PreferredModifierOrder, CSharpOrderModifiersHelper.Instance)
         {
         }
+
+        private const string CS0267 = nameof(CS0267); // The 'partial' modifier can only appear immediately before 'class', 'struct', 'interface', or 'void'
```
Comment (heejaechang) [inline src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersCodeFixProvider.cs]: usually we put const at the top of the class?

Comment (Neme12) [reply src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersCodeFixProvider.cs]: Thanks, I didn't know that. I will update this PR later.

File: src/EditorFeatures/CSharpTest/OrderModifiers/OrderModifiersTests.cs (hunk 2 above)
Comment (Neme12) [reply src/EditorFeatures/CSharpTest/OrderModifiers/OrderModifiersTests.cs]: Tests added in 2366fbc860201049378d0843ff48632fdceb8bfa

File: src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersCodeFixProvider.cs (hunk 3 above)
Comment (Neme12) [reply src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersCodeFixProvider.cs]: Fixed in 2366fbc860201049378d0843ff48632fdceb8bfa

File: src/Features/Core/Portable/OrderModifiers/AbstractOrderModifiersCodeFixProvider.cs [hunk 4]
```diff
@@ -33,8 +33,10 @@ internal abstract class AbstractOrderModifiersCodeFixProvider : SyntaxEditorBase
             _helpers = helpers;
         }
 
-        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
-            ImmutableArray.Create(IDEDiagnosticIds.OrderModifiersDiagnosticId);
+        protected abstract ImmutableArray<string> FixableCompilerErrorIds { get; }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+            => FixableCompilerErrorIds.Add(IDEDiagnosticIds.OrderModifiersDiagnosticId);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/OrderModifiers/AbstractOrderModifiersCodeFixProvider.cs]: i'm a little concerned about this.  Previously, we know the IDE was generating the diagnostic, so we could trust the location of the diagnostic we were then fixing.  We don't control that with the compiler, and so using hte diagnostic without validation can be risky.  Specifically, for code like: 

> var memberDeclaration = diagnostic.Location.FindNode(cancellationToken);

We should actually validate we're on a proper node, and bail out if not.

Comment (Neme12) [reply src/Features/Core/Portable/OrderModifiers/AbstractOrderModifiersCodeFixProvider.cs]: Fixed in 2d3144d46883584277844b7cbf8480fb573739fd

**Comment by sharwell** [comment]
This is cool @Neme12, looking forward to having it 👍

File: src/Features/VisualBasic/Portable/OrderModifiers/VisualBasicOrderModifiersCodeFixProvider.vb (hunk 1 above)
Comment (sharwell) [reply src/Features/VisualBasic/Portable/OrderModifiers/VisualBasicOrderModifiersCodeFixProvider.vb]: I think it makes sense, and allows for future expansion if someone (@MaStr11?) finds VB compiler diagnostics that would be fixed in a similar fashion.

**Pull request merged**